lib/event.c

```diff
@@ -246,25 +246,24 @@ static void cpu_record_clear(uint8_t filter)
 	struct event_loop *m;
 	struct listnode *ln;
 
-	frr_with_mutex (&masters_mtx) {
-		for (ALL_LIST_ELEMENTS_RO(masters, ln, m)) {
-			frr_with_mutex (&m->mtx) {
-				struct cpu_event_history *item;
-				struct cpu_records_head old[1];
-
-				cpu_records_init(old);
-				cpu_records_swap_all(old, m->cpu_records);
-
-				while ((item = cpu_records_pop(old))) {
-					if (item->types & filter)
-						cpu_records_free(&item);
-					else
-						cpu_records_add(m->cpu_records,
-								item);
-				}
-
-				cpu_records_fini(old);
+	frr_mutex_lock_autounlock(&masters_mtx);
```

Comment:
I don't see any changes in this chunk?  What am I missing?

Reply:
yeah, this is one of the patterns that was a little awkward. note that the original had a `with_mutex` block, and then an inner `with_mutex` block. `with_mutex` is a macro, and it uses ... a local variable - so the inner block's variable shadows the outer block's. so the conversion replace the outer block, and that resolves the warning.

Reply:
ooooooh yeah, the variable's inside the macro… (it's warning about `_once`?) - that should be fixable in the header

also note `frr_with_mutex` accepts multiple mutexes and locks them in left-to-right order, albeit tbh I never quite felt comfortable using that since it's very implicit

Reply:
#17937